packages/ragbits-core/docs/how-to/hybrid-search.md

```diff
@@ -0,0 +1,169 @@
+
+This guide explains how to implement hybrid search combining both dense and sparse embeddings in Ragbits.
+
+## What is Hybrid Search?
+
+Hybrid search combines multiple search approaches to get the best of both worlds:
+
+- **Dense embeddings**: Good at capturing semantic meaning and contextual relationships
+- **Sparse embeddings**: Excel at lexical matching and keyword preservation
+
+By combining these approaches, you can create more robust search systems that handle both semantic similarity and exact keyword matching.
+
+## Implementing Hybrid Search in Ragbits
+
+Ragbits supports hybrid search through the `HybridSearchVectorStore` class, which can combine results from multiple vector stores.
+
+### Step 1: Create Dense and Sparse Embedders
+
+First, set up both dense and sparse embedders:
+
+```python
+from ragbits.core.embeddings.base import Embedder
+from ragbits.core.embeddings.sparse import SparseEmbedder
+
+# Dense embedder (example using a pre-configured embedder)
+dense_embedder = Embedder.from_config({
+    "type": "ragbits.core.embeddings.openai.OpenAIEmbedder",
+    "config": {
+        "api_key": "your-api-key",
+        "model": "text-embedding-ada-002"
+    }
+})
+
+# Sparse embedder
+sparse_embedder = SparseEmbedder()
+```
+
+### Step 2: Create Vector Stores for Each Embedder Type
+
+Create separate vector stores for dense and sparse embeddings:
+
+```python
+from ragbits.core.vector_stores.in_memory import InMemoryVectorStore
+from ragbits.core.vector_stores.base import EmbeddingType, VectorStoreOptions
+
+# Vector store for dense embeddings
+dense_store = InMemoryVectorStore(
+    embedder=dense_embedder,
+    embedding_type=EmbeddingType.TEXT,
+    default_options=VectorStoreOptions(k=10)
+)
+
+# Vector store for sparse embeddings
+sparse_store = InMemoryVectorStore(
+    embedder=sparse_embedder,
+    embedding_type=EmbeddingType.TEXT,
+    default_options=VectorStoreOptions(k=10)
+)
+```
+
+### Step 3: Create a Hybrid Search Vector Store
+
+Combine the vector stores using `HybridSearchVectorStore`:
+
+```python
+from ragbits.core.vector_stores.hybrid import HybridSearchVectorStore
+
+hybrid_store = HybridSearchVectorStore(
+    stores=[dense_store, sparse_store],
+    weights=[0.7, 0.3]  # Weights for each store (must sum to 1.0)
+)
+```
+
+### Step 4: Store and Retrieve Entries
+
+Store your entries in both vector stores and then query using the hybrid store:
+
+```python
+import uuid
+from ragbits.core.vector_stores.base import VectorStoreEntry
+
+# Create entries
+entries = [
+    VectorStoreEntry(
+        id=uuid.uuid4(),
+        text="This is a sample document about artificial intelligence",
+        metadata={"category": "AI"}
+    ),
+    VectorStoreEntry(
+        id=uuid.uuid4(),
+        text="Machine learning algorithms can process large datasets",
+        metadata={"category": "ML"}
+    )
+]
+
+# Store entries in both vector stores
+await dense_store.store(entries)
```

Comment:
The whole point of `HybridSearchVectorStore` is that you call it instead of interacting with the indyvidual vector stores directly